src/Governor/FactoryGovernor.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {AccessControl} from "openzeppelin-contracts/access/AccessControl.sol";
+import {GenericFactory} from "evk/GenericFactory/GenericFactory.sol";
+import {RevertBytes} from "evk/EVault/shared/lib/RevertBytes.sol";
+
+import {ReadOnlyProxy} from "./ReadOnlyProxy.sol";
+
+/// @title FactoryGovernor
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice Governor for the EVK beacon (factory), allowing pause guardians to upgrade the implementation to read only
+/// proxy
+contract FactoryGovernor is AccessControl {
+    /// @notice Role identifier for the guardian role.
+    bytes32 public constant GUARDIAN_ROLE = keccak256("GUARDIAN_ROLE");
+
+    /// @notice Event emitted when a factory is paused.
+    /// @param factory The factory that was paused.
+    event Paused(address indexed factory);
+
+    /// @notice Constructor to set the initial admin of the contract.
+    /// @param admin The address of the initial admin.
+    constructor(address admin) {
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
```

Comment:
fair point. I'll change it to use `AccessControlEnumerable` instead of `AccessControl`. `AccessControlDefaultAdminRules` would be even better but it's an overkill in my opinion